electronics_lib/DcDcConverters.py

```diff
@@ -1,5 +1,81 @@
+    ))
+    self.gnd = self.Port(Ground())
+    self.sw = self.Port(VoltageSource())
+    self.fb = self.Port(AnalogSink(impedance=(8000, float('inf')) * kOhm))
+    self.vout = self.Port(Passive()) # TODO model as voltage source
+
+  def contents(self) -> None:
+    super().contents()
+    self.footprint(
+      'U', 'Package_TO_SOT_SMD:SOT-23-6',
+      {
+        '1': self.fb,
+        '2': self.vin, # en
+        '3': self.vin,
+        '4': self.gnd,
+        '5': self.sw,
+        '6': self.vout,
+      },
+      mfr='Texas Instruments', part='TPS61023',
+      datasheet='https://www.ti.com/lit/ds/symlink/tps61023.pdf'
+    )
+
+class Tps61023(DiscreteBoostConverter, GeneratorBlock):
+
+  def contents(self):
+    super().contents()
+    self.assign(self.frequency, (0.5, 1)*MHertz)
+    self.assign(self.efficiency, (0.7, 0.97))
+
+    self.fb = self.Block(FeedbackVoltageDivider(
+      output_voltage=(580, 610) * mVolt,
+      impedance=(1, 300) * kOhm,
+      assumed_input_voltage=self.spec_output_voltage
+    ))
+    self.assign(self.pwr_out.voltage_out,
+                (580*mVolt / self.fb.ratio.upper(),
+                 610*mVolt / self.fb.ratio.lower()))
+
+    self.generator(self.generate_converter,
+                   self.pwr_in.link().voltage, self.spec_output_voltage,
+                   self.pwr_out.link().current_drawn,
+                   self.frequency, self.output_ripple_limit, self.input_ripple_limit, self.ripple_current_factor,
+                   targets=[self.fb, self.pwr_in, self.pwr_out, self.gnd])
+
+
+  def generate_converter(self, input_voltage: RangeVal, output_voltage: RangeVal,
+                         output_current: RangeVal, frequency: RangeVal,
+                         spec_output_ripple: float, spec_input_ripple: float, ripple_factor: RangeVal) -> None:
+    self.ic = self.Block(Tps61023_Device(
+      current_draw=(self.pwr_out.link().current_drawn.lower() * self.pwr_out.voltage_out.lower() / self.pwr_in.link().voltage.upper() / self.efficiency.upper(),
+                    self.pwr_out.link().current_drawn.upper() * self.pwr_out.voltage_out.upper() / self.pwr_in.link().voltage.lower() / self.efficiency.lower())
+    ))
+    self.connect(self.pwr_in, self.ic.vin)
+    self.connect(self.gnd, self.ic.gnd)
+
+    self.connect(self.fb.input, self.pwr_out)
+    self.connect(self.fb.gnd, self.gnd)
+    self.connect(self.fb.output, self.ic.fb)
+
+    self._generate_converter(self.ic.sw, 0.5,
+                             input_voltage=input_voltage, output_voltage=output_voltage,
+                             output_current_max=output_current[1], frequency=frequency,
+                             spec_output_ripple=spec_output_ripple, spec_input_ripple=spec_input_ripple,
+                             ripple_factor=ripple_factor)
+
+    self.connect(self.pwr_out, self.ic.vout.as_voltage_source(
+      voltage_out=self.pwr_out.voltage_out,  # TODO cyclic dependency?
+      current_limits=(0, 1.2)*Amp
```

Comment:
This chip seems to have a higher current limit - "valley switch current limit" of 3.7A. You might also want to update the `rated_current_limit` in `_generate_converter`.

I'm not sure what valley switching current is, but if you have the interest see if you can understand how that translates into the EDG model, and if not, leave it as a TODO.